src/partisan/irods.py

```diff
@@ -1061,12 +1084,19 @@ def __lt__(self, other):
                     return True
 
                 if self.location == other.location:
-                    if self.checksum < other.checksum:
+                    if self.checksum is not None and other.checksum is None:
                         return True
 
-                    if self.checksum == other.checksum:
-                        if self.valid < other.valid:
+                    if self.checksum is None and other.checksum is not None:
+                        return False
+
+                    if self.checksum is not None and other.checksum is not None:
+                        if self.checksum < other.checksum:
```

Comment:
I don't understand comparing checksums in this way, surely they should be checked for correctness (either against metadata, or against the actual object), not just by a string comparison?

Reply:
It's comparing two objects for ordering in a sequence ,rather than saying anything about the correctness of the checksums. We just need to know that if both instances have a checksum, is one "less" than the other.